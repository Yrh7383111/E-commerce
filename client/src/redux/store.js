import { createStore, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import { persistStore } from 'redux-persist';
import rootReducer from './root-reducer';
import createSagaMiddleware from 'redux-saga';
import rootSaga from "./root-saga";



// Saga middleware for Action object after reducer
const sagaMiddleware = createSagaMiddleware();

// Middleware between Action and Reducer
// thunk - catch function
const middlewares = [sagaMiddleware];
if (process.env.NODE_ENV === 'development')
    middlewares.push(logger);

// Store object which holds all the states
export const store = createStore(rootReducer, applyMiddleware(...middlewares));

// Run root saga middleware
sagaMiddleware.run(rootSaga);

// Redux browser Local Storage
export const persistor = persistStore(store);