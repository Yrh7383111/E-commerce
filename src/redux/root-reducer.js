import { combineReducers } from 'redux';
import userReducer from './user/user.reducer';
import cartReducer from "./cart/cart.reducer";



// State - object
// Combine all small Reducers into Root Reducers
export default combineReducers({
    user: userReducer,
    cart: cartReducer
});