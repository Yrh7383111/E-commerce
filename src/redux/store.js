import { createStore, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import { persistStore } from 'redux-persist';
import rootReducer from './root-reducer';



// Middleware between Action and Reducer
const middlewares = [];
if (process.env.NODE_ENV === 'development')
    middlewares.push(logger);

// Store object which holds all the states
export const store = createStore(rootReducer, applyMiddleware(...middlewares));

// Redux browser Local Storage
export const persistor = persistStore(store);