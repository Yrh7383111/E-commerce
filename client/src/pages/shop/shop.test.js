import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { combineReducers, createStore } from 'redux';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import ShopPage from './shop.component';


export const createMockStore = ({ state, reducers }) => {
    const store = createStore(combineReducers(reducers), state);
    return {
        ...store,
        persistor: {
            persist: () => null
        }
    };
};

describe('ShopPage component', () => {
    let wrapper;
    let store;

    beforeEach(() => {
        const mockReducer = (
            state = {
                isFetching: true
            },
            action
        ) => state;
        const mockState = {
            shop: {
                isFetching: true
            }
        };
        store = createMockStore({
            state: mockState,
            reducers: { shop: mockReducer }
        });
        const mockMatch = {
            path: ''
        };
        const mockProps = {
            match: mockMatch,
        };

        wrapper = mount(
            <BrowserRouter>
                <Provider store={store}>
                    <ShopPage {...mockProps} />
                </Provider>
            </BrowserRouter>
        );
    });

    test('should render ShopPage component', () => {
        expect(wrapper).toMatchSnapshot();
    });
});