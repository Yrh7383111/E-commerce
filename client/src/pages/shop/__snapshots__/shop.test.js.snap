// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShopPage component should render ShopPage component 1`] = `ReactWrapper {}`;
